src/icp_canister_backend/tests/test.rs

```diff
@@ -1424,3 +1424,49 @@ fn test_reward_rate_increase_decrease_during_episodes() {
         decreased_reward_rate
     );
 }
+
+#[test]
+fn test_reward_distribution_and_withdrawal() {
+    let (pic, canister_id, ledger_id) = setup();
+    let user = Principal::from_text("xkbqi-2qaaa-aaaah-qbpqq-cai").unwrap();
+    let deposit_amount = Nat::from(100_000_000u64);
+    let reward_amount = Nat::from(50_000_000u64);
+
+    let current_episode = get_stakable_episode(&pic, canister_id, 0);
+    create_deposit(&pic, canister_id, ledger_id, user, deposit_amount.clone(), current_episode);
+
+    reward_pool(&pic, canister_id, ledger_id, user, reward_amount.clone())
+        .expect("Reward pool should succeed");
+
+    advance_time(&pic, 91 * 24 * 60 * 60 / 3 + 1); 
```

Comment:
What is this kind of constant about 91 * 24 * 60 * 60 / 3 + 1?

Reply:
You should always use named constants.
In that case, you probably need call get_episode_time_to_end function.